mula/scheduler/schedulers/schedulers/report.py

```diff
@@ -1,210 +1,125 @@
-        organisation: Organisation,
-        queue: PriorityQueue | None = None,
-        callback: Callable[..., None] | None = None,
-    ):
-        self.logger: structlog.BoundLogger = structlog.get_logger(__name__)
-        self.organisation = organisation
-        self.queue = queue or PriorityQueue(
-            pq_id=scheduler_id,
-            maxsize=ctx.config.pq_maxsize,
-            item_type=self.ITEM_TYPE,
-            allow_priority_updates=True,
-            pq_store=ctx.datastores.pq_store,
-        )
+    """Scheduler implementation for the creation of ReportTask models."""
 
-        super().__init__(
-            ctx=ctx,
-            queue=self.queue,
-            scheduler_id=scheduler_id,
-            callback=callback,
-            create_schedule=True,
-            auto_calculate_deadline=False,
-        )
+    ID: Literal["report"] = "report"
+    TYPE: models.SchedulerType = models.SchedulerType.REPORT
+    ITEM_TYPE: Any = models.ReportTask
+
+    def __init__(self, ctx: context.AppContext):
+        """Initializes the NormalizerScheduler.
+
+        Args:
+            ctx (context.AppContext): Application context of shared data (e.g.
+                configuration, external services connections).
+        """
+        super().__init__(ctx=ctx, scheduler_id=self.ID, create_schedule=True, auto_calculate_deadline=False)
 
     def run(self) -> None:
+        """The run method is called when the schedulers is started. It will
+        start the rescheduling process for the ReportTask models that are
+        scheduled.
+        """
         # Rescheduling
         self.run_in_thread(
-            name=f"scheduler-{self.scheduler_id}-reschedule", target=self.push_tasks_for_rescheduling, interval=60.0
+            name=f"scheduler-{self.scheduler_id}-reschedule", target=self.process_rescheduling, interval=60.0
+        )
+        self.logger.info(
+            "Report scheduler started", scheduler_id=self.scheduler_id, item_type=self.queue.item_type.__name__
         )
 
-    @tracer.start_as_current_span(name="report_push_tasks_for_rescheduling")
-    def push_tasks_for_rescheduling(self):
-        if self.queue.full():
-            self.logger.warning(
-                "Report queue is full, not populating with new tasks",
-                queue_qsize=self.queue.qsize(),
-                organisation_id=self.organisation.id,
-                scheduler_id=self.scheduler_id,
+    @tracer.start_as_current_span(name="process_rescheduling")
+    def process_rescheduling(self):
+        schedules, _ = self.ctx.datastores.schedule_store.get_schedules(
+            filters=filters.FilterRequest(
+                filters=[
+                    filters.Filter(column="scheduler_id", operator="eq", value=self.scheduler_id),
+                    filters.Filter(column="deadline_at", operator="lt", value=datetime.now(timezone.utc)),
+                    filters.Filter(column="enabled", operator="eq", value=True),
```

Comment:
Can we add the correct joins/filters here to limit this result set to only schedules that are allowed to produce a task?

Reply:
These filters should do that. Other evaluations of whether a task should be executed or not depend on the scheduler and subsequent cross-references checks to other services. E.g. for a boefje scheduler, does the ooi still exist, is the plugin still enabled, are we still allowed to scan, etc. These factors can't be evaluated by a database query.

Reply:
Can they be set on the update events of those changes though? Eg, disabling / removing tasks for boefjes that are being disabled seems like the correct way forward instead of keeping them around and checking them for an enabled boefje every time they could be executed.

Reply:
Which is clearly out of scope for this PR, but still worth investigating.